core/domain/play_later_services.py

```diff
@@ -0,0 +1,239 @@
+    return user_domain.PlayLaterActivities(
+        play_later_activities_model.id,
+        play_later_activities_model.exploration_ids,
+        play_later_activities_model.collection_ids)
+
+
+def save_play_later_activities(play_later_activities):
+    """Save an play later activities domain object as an
+    PlayLaterActivitiesModel entity in the datastore.
+
+    Args:
+        play_later_activities: PlayLaterActivities. The play later
+            activities domain object to be saved in the datastore.
+    """
+    play_later_activities_model = user_models.PlayLaterActivitiesModel(
+        id=play_later_activities.id,
+        exploration_ids=(
+            play_later_activities.exploration_ids),
+        collection_ids=play_later_activities.collection_ids)
+
+    play_later_activities_model.put()
+
+
+def mark_exploration_to_be_played_later(
+        user_id, exploration_id, position_to_be_inserted=None):
+    """Adds the exploration id to the play later list of the user at the given
+    position. If the position is not specified, the exploration gets added at
+    the end. If the exploration is already completed or currently being
+    completed it is not added. The maximum limit of the play later list is 10.
+    If the count exceeds 10, the exploration is not added.
+
+    Args:
+        user_id: str. The id of the user.
+        exploration_id: str. The id of the exploration to be added to the play
+            later list.
+        position_to_be_inserted: int|None. If this is specified the exploration
+            gets inserted at the given position. Otherwise it gets added at the
+            end.
+    """
+    # Need to import learner_progress_services so that we are able to fetch the
+    # ids of the completed and incomplete explorations.
+    from core.domain import learner_progress_services
+
+    play_later_activities_model = user_models.PlayLaterActivitiesModel.get(
+        user_id, strict=False)
+    if not play_later_activities_model:
+        play_later_activities_model = (
+            user_models.PlayLaterActivitiesModel(id=user_id))
+
+    subscribed_exploration_ids = (
+        subscription_services.get_exploration_ids_subscribed_to(user_id))
+    completed_exploration_ids = (
+        learner_progress_services.get_all_completed_exp_ids(user_id))
+    incomplete_exploration_ids = (
+        learner_progress_services.get_all_incomplete_exp_ids(user_id))
+
+    play_later_activities = get_play_later_activities_from_model(
+        play_later_activities_model)
+
+    if (exploration_id not in subscribed_exploration_ids and
+            exploration_id not in completed_exploration_ids and
+            exploration_id not in incomplete_exploration_ids):
+        if not position_to_be_inserted:
+            if len(play_later_activities.exploration_ids) < 10:
+                play_later_activities.add_exploration_id(exploration_id)
+        else:
+            play_later_activities.remove_exploration_id(exploration_id)
```

Comment:
Done!

I have checked all the cases in the new code I have added. Also if the insert position is greater the entity will simply get appended at the end of the list. Also since the argument to this function would be added by dragging and dropping entities, the insert position really cannot be greater than the total number of elements.

But even then I've added additional checks in the backend to prevent errors.